pyartifactory/objects.py

```diff
@@ -545,3 +571,148 @@ def delete(self, permission_name: str) -> None:
+                with open(local_filename, "wb") as f:
+                    for chunk in r.iter_content(chunk_size=8192):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                            # f.flush()
+            logging.info(f"Artifact {local_filename} successfully downloaded")
+            return local_filename
+        except ArtifactDownloadException:
+            logging.error(f"Cannot download artifact {local_filename}")
+            raise
+
+    def properties(self, artifact_path: str) -> ArtifactPropertiesResponse:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :return: Artifact properties
+        """
+        artifact_path = artifact_path.lstrip("/")
+        try:
+            r = self._get(f"api/storage/{artifact_path}?properties[=x[,y]]")
+            logging.info("Artifact Properties successfully retrieved")
+            return ArtifactPropertiesResponse(**r.json())
+        except ArtifactPropertiesException:
+            logging.error(f"Cannot retrieve artifact properties")
+            raise
+
+    def stats(self, artifact_path: str) -> ArtifactStatsResponse:
+        """
+        :param artifact_path: Path to file in Artifactory
+        :return: Artifact Stats
+        """
+        artifact_path = artifact_path.lstrip("/")
+        try:
+            r = self._get(f"api/storage/{artifact_path}?stats")
+            logging.info("Artifact stats successfully retrieved")
+            return ArtifactStatsResponse(**r.json())
+        except ArtifactPropertiesException:
+            logging.error(f"Cannot retrieve artifact stats")
+            raise
+
+    def copy(
+        self, artifact_current_path: str, artifact_new_path: str, dryrun: bool = False
+    ) -> bool:
+        """
+        :param artifact_current_path: Current path to file
+        :param artifact_new_path: New path to file
+        :param dryrun: Dry run
+        :return: True if the move is successful
+        """
+        artifact_current_path = artifact_current_path.lstrip("/")
+        artifact_new_path = artifact_new_path.lstrip("/")
+        if dryrun:
+            dry = 1
+        else:
+            dry = 0
+
+        try:
+            self._post(
+                f"api/copy/{artifact_current_path}?to={artifact_new_path}&dry={dry}"
+            )
+            logging.info(f"Artifact {artifact_current_path} successfully copied")
+            return True
+        except ArtifactCopyException:
+            logging.error(f"Cannot copy artifact {artifact_current_path}")
+            raise
+
+    def move(
+        self, artifact_current_path: str, artifact_new_path: str, dryrun: bool = False
+    ) -> bool:
+        """
+        :param artifact_current_path: Current path to file
+        :param artifact_new_path: New path to file
+        :param dryrun: Dry run
+        :return: True if the move is successful

```

Comment:
Same, maybe return the properties for the new location?

Reply:
Same as for the copy, I think that a boolean is better